dts/arm/nuvoton/m46x.dtsi

```diff
@@ -0,0 +1,597 @@
+/*
+ * Copyright (c) 2023 Nuvoton Technology Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <arm/armv7-m.dtsi>
+#include <mem.h>
+#include <zephyr/dt-bindings/gpio/gpio.h>
+#include <zephyr/dt-bindings/i2c/i2c.h>
+#include "m46x/sys.h"
+#include "m46x/sys_reg.h"
+#include "m46x/clk.h"
+#include "m46x/clk_reg.h"
+
+/ {
+	chosen {
+		zephyr,flash-controller = &fmc;
+	};
+
+	cpus {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		cpu@0 {
+			device_type = "cpu";
+			compatible = "arm,cortex-m4f";
+			reg = <0>;
+		};
+	};
+
+	sram0: memory@20000000 {
+		compatible = "mmio-sram";
+		reg = <0x20000000 DT_SIZE_K(512)>;
+	};
+
+	sysclk: system-clock {
+		compatible = "fixed-clock";
+		clock-frequency = <200000000>;
+		#clock-cells = <0>;
+	};
+
+	soc {
+		scc: system-clock-controller@40000200 {
+			compatible = "nuvoton,numaker-scc";
+			reg = <0x40000200 0x100>;
+			#clock-cells = <0>;
+			/* hxt = "enable"; */
+			/* lxt = "enable"; */
+			clk-pclkdiv = <(CLK_PCLKDIV_APB0DIV_DIV2 | CLK_PCLKDIV_APB1DIV_DIV2)>;
+			core-clock = <200000000>;
+
+			pcc: peripheral-clock-controller {
+				compatible = "nuvoton,numaker-pcc";
+				#clock-cells = <3>;
+			};
+		};
+
+		rst: reset-controller@40000000 {
+			compatible = "nuvoton,numaker-rst";
+			reg = <0x40000000 0x20>;
+			#reset-cells = <1>;
+			status = "okay";
+		};
```

Comment:
why is this adding a node without a driver? Please, introduce the relevant nodes in the same commit where you introduce a driver/binding for them.

Reply:
This is especially important since otherwise changes may be needed to the binding/dts node once the driver is written. It is difficult to predict the exact binding needed before writing the driver.

Reply:
Yes, I see, in fact this dtsi was verified by the relevant dirves, the drivers are ready for next PR one by one.
Should I still need to move the node pinctrl/reset/clock/uart/gpio to relevant drive commit one by one ?

Reply:
yes please

Reply:
@henrikbrixandersen & @gmarull,
Thanks of your viewpoint.
Finished the upadte to let node matching driver, please check it.

Reply:
Hi @henrikbrixandersen & @gmarull,
If you are available, please look at this update.